modules/core/02-client/keeper/keeper_test.go

```diff
@@ -483,3 +486,114 @@ func (suite *KeeperTestSuite) TestUnsetParams() {
 		suite.chainA.GetSimApp().IBCKeeper.ClientKeeper.GetParams(ctx)
 	})
 }
+
+// TestScheduleIBCClientUpgrade tests that an IBC client upgrade has been properly scheduled
+func (suite *KeeperTestSuite) TestScheduleIBCClientUpgrade() {
+	var (
+		upgradedClientState *ibctm.ClientState
+		oldPlan, plan       upgradetypes.Plan
+		expError            error
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expPass  bool
+	}{
+		{
+			"valid upgrade proposal",
+			func() {},
+			true,
+		},
+		{
+			"valid upgrade proposal with previous IBC state", func() {
+				oldPlan = upgradetypes.Plan{
+					Name:   "upgrade IBC clients",
+					Height: 100,
+				}
+			}, true,
+		},
+		{
+			"fail: scheduling upgrade with plan height 0",
+			func() {
+				plan.Height = 0
+				expError = sdkerrors.ErrInvalidRequest
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest()  // reset
+			oldPlan.Height = 0 // reset
+
+			path := ibctesting.NewPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupClients(path)
+			upgradedClientState = suite.chainA.GetClientState(path.EndpointA.ClientID).ZeroCustomFields().(*ibctm.ClientState)
+
+			// use height 1000 to distinguish from old plan
+			plan = upgradetypes.Plan{
+				Name:   "upgrade IBC clients",
+				Height: 1000,
+			}
+
+			tc.malleate()
+
+			// set the old plan if it is not empty
+			if oldPlan.Height != 0 {
+				// set upgrade plan in the upgrade store
+				store := suite.chainA.GetContext().KVStore(suite.chainA.GetSimApp().GetKey(upgradetypes.StoreKey))
+				bz := suite.chainA.App.AppCodec().MustMarshal(&oldPlan)
+				store.Set(upgradetypes.PlanKey(), bz)
+
+				bz, err := types.MarshalClientState(suite.chainA.App.AppCodec(), upgradedClientState)
+				suite.Require().NoError(err)
+
+				suite.chainA.GetSimApp().UpgradeKeeper.SetUpgradedClient(suite.chainA.GetContext(), oldPlan.Height, bz) //nolint:errcheck
```

Comment:
may as well check the error instead of having the `nolint` directive, if you always expect it to succeed just use `suite.Require().NoError(err)` on it directly underneath